api/src/main/java/org/openmrs/module/pihcore/htmlformentry/action/AddPatientToQueue.java

```diff
@@ -0,0 +1,75 @@
+package org.openmrs.module.pihcore.htmlformentry.action;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.module.queue.api.QueueServicesWrapper;
+import org.openmrs.module.queue.api.search.QueueSearchCriteria;
+import org.openmrs.module.queue.model.Queue;
+import org.openmrs.module.queue.model.QueueEntry;
+
+public class AddPatientToQueue implements org.openmrs.module.htmlformentry.CustomFormSubmissionAction {
```

Comment:
Let's also add some quick comments of what this class does/doesn't do, see some of my comments below, ie note that it assumes that a service concept maps to one and only one queue.